### PR TITLE
MINOR: Fix ReassignPartitionsClusterTest.testHwAfterPartitionReassignment test

### PR DESCRIPTION
**Pull request by omkreddy**

### CODE REVIEW

**Comment by omkreddy** [comment]
@hachikuji @lindong28 
Observed below test exception on local test runs
```
kafka.admin.ReassignPartitionsClusterTest.testHwAfterPartitionReassignment FAILURE
org.apache.kafka.common.errors.ReplicaNotAvailableException: Replica 100 is not available for partition my-topic-0
```

Pls take a look.

**Comment by lindong28** [comment]
@omkreddy Thanks for the fix. @gitlw reported this issue as well yesterday. The fix LGTM.

File: core/src/test/scala/unit/kafka/admin/ReassignPartitionsClusterTest.scala [hunk 1]
```diff
@@ -98,7 +98,9 @@ class ReassignPartitionsClusterTest extends ZooKeeperTestHarness with Logging {
     )
 
     assertEquals(100, newLeaderServer.replicaManager.getReplicaOrException(topicPartition).highWatermark.messageOffset)
-    servers.foreach(server => waitUntilTrue(() => server.replicaManager.getReplicaOrException(topicPartition).highWatermark.messageOffset == 100, ""))
+    val replicaServer = servers.find(_.config.brokerId == 102).get
```
Comment (lindong28) [inline core/src/test/scala/unit/kafka/admin/ReassignPartitionsClusterTest.scala]: nits: how about newFollowerServer?

**Review by hachikuji — APPROVED** [review]
LGTM. Thanks for fixing this!

**Pull request merged**